wom/src/main/scala/wdl/expression/TypeEvaluator.scala

```diff
@@ -83,10 +83,13 @@ case class TypeEvaluator(override val lookup: String => WdlType, override val fu
           evaluate(a.getAttribute("lhs")).flatMap {
             case o: WdlCallOutputsObjectType =>
               o.call.outputs.find(_.unqualifiedName == rhs.getSourceString) match {
-                case Some(taskOutput) => 
-                  from map { source =>
-                    evaluate(taskOutput.requiredExpression.ast) map { t => DeclarationInterface.relativeWdlType(source, taskOutput, t) }
-                  } getOrElse evaluate(taskOutput.requiredExpression.ast)
+                case Some(taskOutput) =>
+                  val t = taskOutput.wdlType
```

Comment:
So this is using the declared output type rather than the type of the output expression? I like that - the output should be evaluated and then coerced into the declared type so IMO this is more accurate!

Reply:
Isn't the caller perhaps trying to ascertain if the expression actually can be coerced to the declared type?